### PR TITLE
Scripts - Add Hunt Script

### PR DESCRIPTION
**Pull request by TyroneMF**
- Script that has an AI Group hunt a predefined player group, or the closest one within 1km distance.
- Can be used on groups performing waypoints, if the hunt is successful they will return to waypoints.

### CODE REVIEW

File: Scripts/fn_hunt.sqf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        private _huntedPos = _huntedLeader getPos [random 100, random 360];
+        _hunters move _huntedPos;
+    };
+
+    // Check for alive units
+    private _huntersDead = {alive _x} count units _hunters == 0;
+    private _huntedDead = {alive _x} count units _hunted == 0;
+
+    // Remove PFH.
+    if (_huntersDead || _huntedDead && !isNull _hunted) then {
```
Comment (veteran29) [inline Scripts/fn_hunt.sqf:62]:
```suggestion
    if (_huntersDead || {_huntedDead && !isNull _hunted}) then {
```

**Review by veteran29 — CHANGES_REQUESTED** [review]

File: Scripts/fn_hunt.sqf [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        _hunted = group _nearest;
+        _args set [2, _hunted];
+    };
+    if (!isNull _hunted) then {
+        // Get Group Leaders
+        private _huntedLeader = leader _hunted;
+
+        // Move to estimated hunted leader position
+        private _huntedPos = _huntedLeader getPos [random 100, random 360];
+        _hunters move _huntedPos;
```
Comment (veteran29) [inline Scripts/fn_hunt.sqf:59]: `move` needs local argument, what if the group is already on the HC when the script is executed?

**Review by veteran29 — CHANGES_REQUESTED** [review]

File: Scripts/fn_hunt.sqf (hunk 2 above)
Comment (TyroneMF) [reply Scripts/fn_hunt.sqf:59]: `_hunters setVariable ["acex_headless_blacklist", true];`

Comment (jonpas) [reply Scripts/fn_hunt.sqf:59]: But that's called from a trigget. @veteran29 is asking about AI that is already on the HC when this script runs.

Comment (TyroneMF) [reply Scripts/fn_hunt.sqf:59]: Does the blacklist not pull them back off of it?

Comment (jonpas) [reply Scripts/fn_hunt.sqf:59]: No, it will just keep it on HC it is on forever, preventing moving it again (if any other HC connects/disconnects)

Comment (jonpas) [reply Scripts/fn_hunt.sqf:59]: If you want to move it back, you want to do `GROUP setGroupOwner 2` (2 is server) after blacklisting, on the server.

Also blacklisting should be public `setVariable` then.

Comment (TyroneMF) [reply Scripts/fn_hunt.sqf:62]: Done both.

**Review by jonpas — APPROVED** [review]

**Pull request merged**